src/sort_filter/KalmanTracker.cpp

```diff
@@ -0,0 +1,117 @@
+///////////////////////////////////////////////////////////////////////////////
+// KalmanTracker.cpp: KalmanTracker Class Implementation Declaration
+
+#include "KalmanTracker.h"
+#include <ctime>
+
+using namespace std;
+using namespace cv;
+
+int KalmanTracker::kf_count = 0;
+
+// initialize Kalman filter
+void KalmanTracker::init_kf(
+	StateType stateMat)
+{
+	int stateNum = 7;
+	int measureNum = 4;
+	kf = KalmanFilter(stateNum, measureNum, 0);
+
+	measurement = Mat::zeros(measureNum, 1, CV_32F);
+
+	kf.transitionMatrix = (Mat_<float>(stateNum, stateNum) << 1, 0, 0, 0, 1, 0, 0,

```

Comment:
Code-resiliency question here: If you're hardcoding the initialization of the matrix, shouldn't you hardcode the dimension as well? If someone changes `stateNum` to, say, `5` this will break, right? Because it's hardcoded to be `7x7`. I think if you're going to hardcode the matrix for a specific size, it's actually cleaner to hardcode that size as well:
```suggestion
	kf.transitionMatrix = (Mat_<float>(7, 7) << 1, 0, 0, 0, 1, 0, 0,

```
...Otherwise, best to have the code generate a properly-dimensioned matrix programmatically, based on the value of `stateNum`. Maintaining a variable separate from the hardcoded initialization based on the (assumed) value of that variable feels like a maintenance headache.

Reply:
(`stateNum` can be set to `7` _after_ the hardcoded matrix initialization, of course. As long as there's a comment noting that its value needs to be equal to the size of either dimension of the square `transitionMatrix`.)